tests/integration/command-functions.test.ts

```diff
@@ -103,69 +138,111 @@ describe('Process creation and deletion, listing', () => {
+          const res = await loadProcesses({
+            options: polkadotOptions,
+            data: multiple('existing-length', 1, process2Name, process2BumpedV),
+          })
+
+          expect(res['existing-length'].message).to.equal('existing: programs are different lengths')
+          expect(res[process2Name].message).to.equal('Transaction for new process should-create-1 has been successfully submitted')
+
+        })
+
+        it('also fails if number of steps matches but POSTFIX does not', async () => {
+          await createProcess('existing-steps', 1, simple, false, polkadotOptions)
+          const process2Name = 'should-create-2'
+          const process2BumpedV = (await getVersionHelper(process2Name)) + 1
+          const res = await loadProcesses({
+            options: polkadotOptions,
+            data: multiple('existing-steps', 1, process2Name, process2BumpedV),
+          })
+
+          expect(res['existing-steps'].message).to.equal('existing: program steps did not match')
+          expect(res[process2Name].message).to.equal('Transaction for new process should-create-2 has been successfully submitted')
+        })
+      })
+
+      it('does not create new one and notifies if programs are different length', async () => {
+        await createProcess('existing-single', 1, validAllRestrictions, false, polkadotOptions)
+        const { message, process } = await createProcess('existing-single', 1, simple, false, polkadotOptions)
+
+        expect(message).to.equal('existing: programs are different lengths')
+        expect(process).to.deep.contain({
+          id: '0x6578697374696e672d73696e676c65',
+          version: 1,
+          status: 'Enabled',
+        })
+      })
+
+      it('does not create new one and notifies if programs same are length but do not match', async () => {
+        await createProcess('existing-steps-single', 1, simple2, false, polkadotOptions)
+        const { message, process } = await createProcess('existing-steps-single', 1, [{ restriction: { None: {} }}], false, polkadotOptions)
+
+        expect(message).to.equal('existing: program steps did not match')
+        expect(process).to.deep.contain({
+          id: '0x6578697374696e672d73746570732d73696e676c65',
+          version: 1,
+          status: 'Enabled',
+        })
+      })
+    })
+
+    it('fails for invalid POSTFIX notation', async () => {
+      const { message } = await createProcess(validProcessName, validVersionNumber, invalidPOSIX, false, polkadotOptions)
+      expect(message).to.equal('invalid program')
     })
 
     it('fails for invalid restriction key', async () => {
-      return assert.isRejected(
-        createProcess(validProcessName, validVersionNumber, invalidRestrictionKey, false, polkadotOptions),
-        ZodError
-      )
+      const { message } = await createProcess(validProcessName, validVersionNumber, invalidPOSIX, false, polkadotOptions)
+      expect(message).to.equal('invalid program')
```

Comment:
this is completely changing what's tested? It's no longer testing for an invalid restriction key

Reply:
actually, hold on a sec, should be able to swap back after returning err as a whole